### PR TITLE
enable search on token entities

### LINKED ISSUE #7705: Enable search filter on Assethubs
### What happened?

Search filter does not work: 

image

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix

## Context

- [x] Closes #7705
- [x] Requires deployment <stick>
https://github.com/kodadot/stick/pull/121

* cahnge from draft after deployment

### <samp>🤖 Generated by Copilot at a1ccc0c</samp>

Added a new feature to search tokens by name in a case-insensitive way. Refactored the code that generates the search criteria for the `tokenListWithSearch` query in `useItemsGrid.ts`.

### <samp>🤖 Generated by Copilot at a1ccc0c</samp>

> _To search for some tokens by name_
> _We added a new query param_
> _`name_containsInsensitive`_
> _Makes the filter more permissive_
> _And `getSearchCriteria` is more tame_

### CODE REVIEW

**Review by Jarsen136 — APPROVED** [review]

File: queries/subsquid/general/tokenListWithSearch.graphql [hunk 1]
```diff
@@ -9,6 +9,7 @@ query tokenListWithSearch(
   $issuer: String
   $denyList: [String!]
   $collections: [String!]
+  $name_containsInsensitive: String
```
Comment (roiLeo) [inline queries/subsquid/general/tokenListWithSearch.graphql:12]:
```suggestion
  $name: String
```

File: queries/subsquid/general/tokenListWithSearch.graphql [hunk 2]
```diff
@@ -21,6 +22,7 @@ query tokenListWithSearch(
     price_gt: $price_gt
     price_lte: $price_lte
     collections: $collections
+    name_containsInsensitive: $name_containsInsensitive
```
Comment (roiLeo) [inline queries/subsquid/general/tokenListWithSearch.graphql:25]:
```suggestion
    name: $name
```

File: queries/subsquid/general/tokenListWithSearch.graphql [hunk 3]
```diff
@@ -52,6 +54,7 @@ query tokenListWithSearch(
...
```
Comment (roiLeo) [inline queries/subsquid/general/tokenListWithSearch.graphql:57]:
```suggestion
    name: $name
```

File: components/items/ItemsGrid/useItemsGrid.ts [hunk 4]
```diff
@@ -45,20 +45,20 @@ export function useFetchSearch({
...
 
   const getSearchCriteria = (searchParams) => {
+    const mapping = {
+      currentOwner_eq: (value) => ({ owner: value }),
+      issuer_eq: (value) => ({ issuer: value }),
+      price_gt: (value) => ({ price_gt: Number(value) }),
+      collection: (value) => ({ collections: value.id_in }),
+      name_containsInsensitive: (value) => ({
+        name_containsInsensitive: value,
+      }),
```
Comment (roiLeo) [inline components/items/ItemsGrid/useItemsGrid.ts:55]:
```suggestion
      name: (value) => ({
        name: value,
      }),
```

**Comment by daiagi** [comment]
works now

**Comment by daiagi** [comment]
@prury seems like test needs to be updated to include stacked cards on profile page

File: components/items/ItemsGrid/useItemsGrid.ts (hunk 4 above)
Comment (roiLeo) [reply components/items/ItemsGrid/useItemsGrid.ts]: well it's the same

**Review by roiLeo — APPROVED** [review]

**Pull request merged**